src/lib/Embed.hs

```diff
@@ -256,6 +264,10 @@ getSnd :: MonadEmbed m => Atom -> m Atom
 getSnd (PairVal _ y) = return y
 getSnd p = emitOp $ Snd p
 
+getUnpacked :: MonadEmbed m => Atom -> m [Atom]
+getUnpacked (Record items) = return $ toList items
+getUnpacked a = emitUnpack (Atom a)
```

Comment:
Well single-case ADTs can also be unpacked, so this might end up being a little confusing.

Reply:
And it seems like there's only a single use-site, so it's better to inline this.

Reply:
I think the fallback case makes it so that this will still work for ADTs, but it just doesn't have the opportunistic optimization (which would be easy to add). Although I'm happy to just inline this. Wasn't sure what the bar was for adding helper functions to Embed (maybe it's two or three use-sites?).